crates/turborepo-lockfiles/src/bun/id.rs

```diff
@@ -0,0 +1,48 @@
+/// An iterator that returns increasingly less specific keys
+pub struct PossibleKeyIter<'a> {
+    key: Option<&'a str>,
+}
+
+impl<'a> PossibleKeyIter<'a> {
+    pub fn new(key: &'a str) -> Self {
+        Self { key: Some(key) }
+    }
+}
+
+impl<'a> Iterator for PossibleKeyIter<'a> {
+    type Item = &'a str;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        let curr = self.key;
+        self.key = curr.and_then(less_specific_key);
+        curr
+    }
+}
+
+fn less_specific_key(key: &str) -> Option<&str> {
+    let slash_idx = key.find('/')?;
```

Comment:
tiny nit: [`split_once`](https://doc.rust-lang.org/std/primitive.str.html#method.split_once) might work better here

Reply:
This is one of those cases where might need a `str` that includes the `/` I split on so I feel like `find` is a little easier to work with here.